iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergDeleteOrphanFiles.java

```diff
@@ -0,0 +1,232 @@
+    LOG.debug("Valid content file for {} are {}", table.name(), validFiles);
+    try {
+      Path dataPath = new Path(tableLocation, DATA_FOLDER_NAME);
+      return getFilesToBeDeleted(lastTime, validFiles, dataPath);
+    } catch (IOException ex) {
+      throw new UncheckedIOException(ex);
+    }
+  }
+
+  public Set<String> getAllContentFilePath() {
+    Set<String> validFilesPath = Sets.newHashSet();
+    Table metadatTable = getMetadataTable();
+
+    TableScan tableScan = metadatTable.newScan();
+    CloseableIterable<FileScanTask> manifestFileScanTasks = tableScan.planFiles();
+    CloseableIterable<StructLike> entries = CloseableIterable.concat(entriesOfManifest(manifestFileScanTasks));
+
+    for (StructLike entry : entries) {
+      StructLike fileRecord = entry.get(4, StructLike.class);
+      String filePath = fileRecord.get(1, String.class);
+      validFilesPath.add(getUriPath(filePath));
+    }
+    return validFilesPath;
+  }
+
+  private Iterable<CloseableIterable<StructLike>> entriesOfManifest(
+      CloseableIterable<FileScanTask> fileScanTasks) {
+    return Iterables.transform(
+        fileScanTasks,
+        task -> {
+          assert task != null;
+          return ((DataTask) task).rows();
+        });
+  }
+
+  public static Set<String> getAllStatisticsFilePath(Table table) {
+    return ReachableFileUtil.statisticsFilesLocations(table).stream().map(HiveIcebergDeleteOrphanFiles::getUriPath)
+        .collect(Collectors.toSet());
+  }
+
+  protected Set<String> cleanMetadata(long lastTime) {
+    LOG.info("{} start clean metadata files", table.name());
+    try {
+      Set<String> validFiles = getValidMetadataFiles(table);
+      LOG.debug("Valid metadata files for {} are {}", table.name(), validFiles);
+      Path metadataLocation = new Path(tableLocation, METADATA_FOLDER_NAME);
+      return getFilesToBeDeleted(lastTime, validFiles, metadataLocation);
+    } catch (IOException ioe) {
+      throw new UncheckedIOException(ioe);
+    }
+  }
+
+  private Set<String> getFilesToBeDeleted(long lastTime, Set<String> validFiles, Path location)
+      throws IOException {
+    Set<String> filesToDelete = Sets.newHashSet();
+    FileSystem fs = location.getFileSystem(conf);
+    RemoteIterator<LocatedFileStatus> metadataLocations = fs.listFiles(location, true);
+    while (metadataLocations.hasNext()) {
+      LocatedFileStatus metadataFile = metadataLocations.next();
+      if (metadataFile.getModificationTime() < lastTime && !validFiles.contains(
+          getUriPath(metadataFile.getPath().toString()))) {
+        filesToDelete.add(metadataFile.getPath().toString());
+      }
+    }
+    return filesToDelete;
+  }
+
+  private Table getMetadataTable() {
+    return MetadataTableUtils.createMetadataTableInstance(((HasTableOperations) table).operations(), table.name(),
+        table.name() + "#" + ALL_ENTRIES.name(), ALL_ENTRIES);
+  }
+
+  private static Set<String> getValidMetadataFiles(Table internalTable) {
```

Comment:
What's the meaning of `internalTable`?

Reply:
I changed it icebergTable, no special meaning, it is the normal table, not sure why I did that :-)